cmd/peer-rest-server.go

```diff
@@ -1097,6 +1097,33 @@ func (s *peerRESTServer) GetPeerMetrics(w http.ResponseWriter, r *http.Request)
 	w.(http.Flusher).Flush()
 }
 
+// EnqueueTransitionTask enqueues an ILM transition object task for an object
+// version specified via query pararms.
+func (s *peerRESTServer) EnqueueTransitionTask(w http.ResponseWriter, r *http.Request) {
+	if !s.IsValid(w, r) {
+		s.writeErrorResponse(w, errors.New("invalid request"))
+		return
+	}
+
+	params := r.URL.Query()
+	bucket := params.Get(peerRESTBucket)
+	object := params.Get(peerRESTObject)
+	versionID := params.Get(peerRESTVersionID)
+	objAPI := newObjectLayerFn()
+	opts := ObjectOptions{
```

Comment:
```suggestion
	objAPI := newObjectLayerFn()
	opts := ObjectOptions{
```'

Check for `objAPI == nil`  for safety